### PR TITLE
docs: Add clarification about `eslint-enable`

### PR DESCRIPTION
**Pull request by dosisod**
Closes https://github.com/eslint/eslint/issues/15662

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Add clarification to how the `eslint-enable` rule works, and what to do if you want to guarantee a rule is always ignored.

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

File: docs/user-guide/configuring/rules.md [hunk 1]
```diff
@@ -160,6 +162,14 @@ You can also disable or enable specific rules for an entire file:
 alert('foo');
 ```
 
+To ensure that a rule is always applied (regardless of any future enable/disable lines):
+
+```js
+/* eslint no-alert: "off" */
+
+alert('foo');
+```
```
Comment (mdjermanovic) [inline docs/user-guide/configuring/rules.md:171]: Isn't it the opposite, this ensures that a rule is _not_ applied regardless of enable/disable comments?

**Review by mdjermanovic — CHANGES_REQUESTED** [review]

File: docs/user-guide/configuring/rules.md (hunk 1 above)
Comment (dosisod) [reply docs/user-guide/configuring/rules.md:171]: You're right, I'll go ahead and update that now

File: docs/user-guide/configuring/rules.md [hunk 2]
```diff
@@ -144,6 +144,8 @@ console.log('bar');
 /* eslint-enable no-alert, no-console */
 ```
 
+**Note:** `/* esline-enable */` without any specific rules will cause all errors to be re-enabled. You will have to re-specify the errors you wish to keep disabled.
```
Comment (nzakas) [inline docs/user-guide/configuring/rules.md:147]:
```suggestion
**Note:** `/* esline-enable */` without any specific rules listed will cause all disabled rules to be re-enabled.
```

**Review by nzakas — APPROVED** [review]
Thanks!

File: docs/user-guide/configuring/rules.md [hunk 3]
```diff
@@ -144,6 +144,8 @@ console.log('bar');
 /* eslint-enable no-alert, no-console */
 ```
 
+**Note:** `/* esline-enable */` without any specific rules listed will cause all disabled rules to be re-enabled.
```
Comment (mdjermanovic) [inline docs/user-guide/configuring/rules.md:147]:
```suggestion
**Note:** `/* eslint-enable */` without any specific rules listed will cause all disabled rules to be re-enabled.
```

Typo

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**